Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -380,7 +380,18 @@ extension BrowserViewController: WKNavigationDelegate {
       if let documentTargetURL = documentTargetURL {
         let domainForShields = Domain.getOrCreate(forUrl: documentTargetURL, persistent: !isPrivateBrowsing)
         let isScriptsEnabled = !domainForShields.isShieldExpected(.NoScript, considerAllShieldsOption: true)
+        
+        // Due to a bug in iOS WKWebpagePreferences.allowsContentJavaScript does NOT work!
+        // https://github.com/brave/brave-ios/issues/8585
+        //
+        // However, the deprecated API WKWebViewConfiguration.preferences.javaScriptEnabled DOES work!
+        // Even though `configuration` is @NSCopying, somehow this actually updates the preferences LIVE!!
+        // This follows the same behaviour as Safari
+        //
+        // - Brandon T.
+        //
         preferences.allowsContentJavaScript = isScriptsEnabled
+        webView.configuration.preferences.javaScriptEnabled = isScriptsEnabled
```

Comment:
Does setting it via https://developer.apple.com/documentation/webkit/wkwebviewconfiguration/3194420-defaultwebpagepreferences work which is not deprecated?

Reply:
Another thing to check: Does creating a new `WKWebpagePreferences` object instead and returning that instead of assigning `allowsContentJavaScript` to the one passed into the delegate method work?

Reply:
Neither of those work.